src/main/js/MergeForm.tsx

```diff
@@ -63,33 +69,38 @@ class MergeForm extends React.Component<Props> {
       shouldDeleteSourceBranch,
       onChangeDeleteSourceBranch,
       onResetCommitMessage,
-      t
+      t,
+      commitMessageHint
     } = this.props;
 
+    const commitMessageElement = <>
+      <Textarea
+        placeholder={t("scm-review-plugin.showPullRequest.mergeModal.commitMessage")}
+        disabled={this.isCommitMessageDisabled()}
+        value={commitMessage}
+        onChange={onChangeCommitMessage}
+      />
+      {this.isShowMessageHint() && (
+        <CommitMessageInfo className="is-size-7">
+            <span className="icon is-small has-text-info">
+              <i className="fas fa-info-circle"/>
+            </span>{" "}
+          <span>{t('scm-review-plugin.showPullRequest.mergeModal.commitMessageHint.' + commitMessageHint)}</span>
+        </CommitMessageInfo>
+      )}
+      <Button label={t("scm-review-plugin.showPullRequest.mergeModal.resetMessage")} action={onResetCommitMessage}/>
+      <hr/>
+    </>
+
     return (
       <>
         <MergeStrategies
           strategyLinks={strategyLinks}
           selectedStrategy={selectedStrategy}
           selectStrategy={selectStrategy}
         />
-        <hr />
-        <Textarea
-          placeholder={t("scm-review-plugin.showPullRequest.mergeModal.commitMessage")}
-          disabled={this.isCommitMessageDisabled()}
-          value={commitMessage}
-          onChange={onChangeCommitMessage}
-        />
-        {this.isShowMessageHint() && (
-          <CommitMessageInfo className="is-size-7">
-            <span className="icon is-small has-text-info">
-              <i className="fas fa-info-circle" />
-            </span>{" "}
-            <span>{t("scm-review-plugin.showPullRequest.mergeModal.fallbackMessageInfo")}</span>
-          </CommitMessageInfo>
-        )}
-        <Button label={t("scm-review-plugin.showPullRequest.mergeModal.resetMessage")} action={onResetCommitMessage} />
-        <hr />
+        <hr/>
+        {this.isCommitMessageVisible() && commitMessageElement}
```

Comment:
You could add the check `this.isCommitMessageVisible() && ...` in the assignment of `commitMessageElement`, so that this component will not be built at all when it is not needed.

Reply:
Great Idea! :)